hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1187,37 +1186,68 @@ public int clear() {
 
     @Override
     public void reset() {
-        clearMapStore();
-        storage.clear(false);
+        clearPartition(false, false);
```

Comment:
This will basically do:
```
        clearMapStore();
        clearLockStore();
        clearIndexedData(false);
        storage.clear(isDuringShutdown);
        mutationObserver.onClear();
```
The problem here is that we are clearing the data from the lock store and calling `onClear` for the mutation observers, one of which is the event journal.
During `reset`, each service should clear only the data which it owns. This means that the `LockService` is responsible for clearing locks and the `RingbufferService` is resposible for clearing event journals.

I also see this method is now calling `clearIndexedData`. Was that the leak?

Reply:
good catch, reverted that part.